File: scratch/refactor_imports.js
const fs = require('fs');
const path = require('path');

const appDir = 'apps/web/app';

// Find all page.tsx files
const pages = [];
function findPages(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            findPages(fullPath);
        } else if (file === 'page.tsx') {
            pages.push(fullPath);
        }
    }
}
findPages(appDir);

for (const page of pages) {
    let content = fs.readFileSync(page, 'utf8');
    
    // Replace all matching imports
    content = content.replace(/from "\.\.\/\.\.\/\.\.\/modules\/[^"]+"/g, 'from "@workspace/marketing"');
    content = content.replace(/from "\.\.\/modules\/core\/components\/layout\/header"/g, 'from "@workspace/marketing"');
    content = content.replace(/from "\.\.\/modules\/core\/components\/layout\/footer"/g, 'from "@workspace/marketing"');
    content = content.replace(/from "\.\.\/modules\/home\/components\/hero"/g, 'from "@workspace/marketing"');
    content = content.replace(/from "\.\.\/modules\/home\/components\/social-proof"/g, 'from "@workspace/marketing"');
    content = content.replace(/from "\.\.\/modules\/home\/components\/features-grid"/g, 'from "@workspace/marketing"');
    content = content.replace(/from "\.\.\/modules\/home\/components\/interactive-demo"/g, 'from "@workspace/marketing"');
    content = content.replace(/from "\.\.\/\.\.\/modules\/[^"]+"/g, 'from "@workspace/marketing"');

    fs.writeFileSync(page, content);
}

File: scratch/refactor_imports.py
import os
import glob
import re

app_dir = "apps/web/app"
pages = glob.glob(f"{app_dir}/**/page.tsx", recursive=True)

for page in pages:
    with open(page, "r") as f:
        content = f.read()
    
    # Replace any relative imports containing /modules/ with @workspace/marketing
    # e.g., "../../../modules/contact/components/contact-info" -> "@workspace/marketing"
    new_content = re.sub(r'from "\.\./\.\./\.\./modules/[^"]+"', 'from "@workspace/marketing"', content)
    new_content = new_content.replace('from "../modules/core/components/layout/header"', 'from "@workspace/marketing"')
    new_content = new_content.replace('from "../modules/core/components/layout/footer"', 'from "@workspace/marketing"')
    new_content = new_content.replace('from "../modules/home/components/hero"', 'from "@workspace/marketing"')
    new_content = new_content.replace('from "../modules/home/components/social-proof"', 'from "@workspace/marketing"')
    new_content = new_content.replace('from "../modules/home/components/features-grid"', 'from "@workspace/marketing"')
    new_content = new_content.replace('from "../modules/home/components/interactive-demo"', 'from "@workspace/marketing"')

    
    with open(page, "w") as f:
        f.write(new_content)
